### PR TITLE
chore: move to single eslintrc file

### PR DESCRIPTION
**Pull request by straker**
I was actually surprised by how many eslintrc files we had. Looking over the ones we had, I think this is the merge of them all, but I want @stephenmathieson to look over our one config to see if the rule set is good (and if the config is good too).

Closes issue: #2156

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

### CODE REVIEW

**Comment by stephenmathieson** [comment]
This PR makes me happy 😁

At first glance, this LGTM but I'll take a more in-depth look tomorrow morning.

**Review by stephenmathieson — CHANGES_REQUESTED** [review]
Couple of small things.

Also, what do you think about renaming `.eslintrc` to `.eslinrc.json` or `.eslintrc.js`? This would enable syntax highlighting and code formatting.

File: .eslintrc [hunk 1]
```diff
@@ -78,6 +80,43 @@
...
+      },
+      "globals": {
+        "describe": true,
+        "it": true,
+        "xit": true,
+        "before": true,
+        "beforeEach": true,
+        "after": true,
+        "afterEach": true,
+        "chai": true,
```
[the remark this answers is not in the document]
Comment (straker) [reply .eslintrc:112]: Chai shouldn't be, but `assert` needs to be since there is no `chai` env

**Review by stephenmathieson — APPROVED** [review]

File: .eslintrc (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephenmathieson) [reply .eslintrc]: I assume that `assert` is actually global in the tests then?

[the remark this answers is not in the document]
Comment (straker) [reply .eslintrc]: ya, it comes from chai (which all the tests load implicitly through karma or explicitly in the integration test pages through `script` tag)

**Pull request merged**